### ISSUE TITLE
 Support for Unicode 13

### ISSUE THREAD

**Issue by codic12** [C1710/blobmoji#58]
Any idea when this will be implemented?

**Comment by C1710** [comment]
Probably not before June or maybe even later, because I'm currently busy with my Bachelor's thesis 😅

**Comment by gsbhasin123** [comment]
Oh, great!
No worries, and good luck!
:)

**Comment by Madis0** [comment]
Good news, some (most?) new emoji can be taken directly from Android 11 due to a change of style. https://9to5google.com/2020/06/21/android-11-turtle-emoji/

**Comment by C1710** [comment]
> 
> 
> Good news, most animal emoji can be taken directly from Android 11 due to a change of style. https://9to5google.com/2020/06/21/android-11-turtle-emoji/

That's indeed interesting (and TBH a bit surprising), but nice.  
However it's not officially published yet, so I'll wait at least until https://github.com/googlefonts/noto-emoji is updated.

**Comment by Madis0** [comment]
Newer article, FWIW https://9to5google.com/2020/07/16/final-android-11-emoji/

**Comment by C1710** [comment]
> 
> 
> Newer article, FWIW https://9to5google.com/2020/07/16/final-android-11-emoji/

Yeah, I've read about it, although it doesn't seem to be a part of their repository yet.

**Comment by C1710** [comment]
Oh, looks like there has been an update almost two weeks ago: https://github.com/googlefonts/noto-emoji  
(whoops)

**Comment by C1710** [comment]
I'll probably start porting to Emoji 13.1 (including 12.1 and 13.0 of course) in 2-3 weeks.  
It will probably include these new emojis from Google (along with some older ones. The current foot and leg emoji are just awful and) and it will be the first release using the new build tool (including a new flag style and a single color palette that's used for all emojis).

**Comment by C1710** [comment]
So, the [smiling face with tears](https://emojipedia.org/smiling-face-with-tear/) is now officially the first new emoji of Emoji 13 that has been added after the standard got the released (technically the transgender pride flag was the first one, but at that moment it hasn't been standardized yet):  
https://github.com/C1710/blobmoji/blob/emoji13/svg13/emoji_u1f972.svg

Just FYI: I will probably not add pre-rendered PNGs anymore (maybe once everything is ready).

**Comment by Madis0** [comment]
https://blog.emojipedia.org/android-11-0-emoji-changelog/

**Comment by C1710** [comment]
There are now custom-made emojis for:  
Coin - https://github.com/C1710/blobmoji/blob/b74773c28ed57e5442606fd1894f0006449939c8/svg13/coin.svg  
Disguised face (might be controversial): https://github.com/C1710/blobmoji/blob/cdc02df5fd9bfc25537debdd44931bae357a4f1b/svg13/disugised%20face.svg  
Headstone (this is more a pun than an emoji): https://github.com/C1710/blobmoji/blob/cdc02df5fd9bfc25537debdd44931bae357a4f1b/svg13/headstone.svg

**Comment by BlackRabbit22** [comment]
Thank you :)

**Comment by AeonCaptain** [comment]
I love the tombstone one. (Is it just me or is that their rightful place. Long live the blobs.)

**Comment by Zodigram** [comment]
I'm not sure where else to say this, but thank you so much for keeping the blobs alive

**Comment by choukawaii** [comment]
Still no sign for the release...

**Comment by DavidBerdik** [comment]
If you are willing to sacrifice consistency for completeness, you can use my Blobmoji build here: https://github.com/DavidBerdik/blobmoji-font/releases/tag/blobmoji-13.1-pre

**Comment by Measureless** [comment]
> Still no sign for the release...

It looks like falling on deaf ears...

**Comment by DavidBerdik** [comment]
> > Still no sign for the release...
> 
> It looks like falling on deaf ears...

@Measureless To be fair, this is just a side project for the developer. I too have side projects that sometimes fall behind because life gets too busy. Give the developer a chance to catch up with whatever is going on in their personal life. I am sure they will be back at it as soon as they can be! As much as the blob emojis are amazing, there are more important things in life! And of course, there is already an incomplete but usable alternative for getting the entire Unicode 13.1 emoji set available here: https://github.com/DavidBerdik/blobmoji-font/releases/tag/blobmoji-13.1-pre

**Comment by C1710** [comment]
So, the Emoji13 branch should now contain all the assets for Emoji 13 (except for gendered mixed-skintone couples with the person with the lighter skincolor standing left).  
But, I currently have an issue with the rendering, so it'll probably take at least until tomorrow, plus the EmojiCompat version which will be released a bit later.

**Comment by Measureless** [comment]
> > > Still no sign for the release...
> > 
> > 
> > It looks like falling on deaf ears...
> 
> @Measureless To be fair, this is just a side project for the developer. I too have side projects that sometimes fall behind because life gets too busy. Give the developer a chance to catch up with whatever is going on in their personal life. I am sure they will be back at it as soon as they can be! As much as the blob emojis are amazing, there are more important things in life! And of course, there is already an incomplete but usable alternative for getting the entire Unicode 13.1 emoji set available here: https://github.com/DavidBerdik/blobmoji-font/releases/tag/blobmoji-13.1-pre

I am gonna try~

**Comment by C1710** [comment]
> 
> 
> So, the Emoji13 branch should now contain all the assets for Emoji 13 (except for gendered mixed-skintone couples with the person with the lighter skincolor standing left).
> But, I currently have an issue with the rendering, so it'll probably take at least until tomorrow, plus the EmojiCompat version which will be released a bit later.

Just FYI, this is how the emojis are currently rendered and I have absolutely no idea, why:
![emoji_u1f4a6](https://user-images.githubusercontent.com/10349490/122639026-e9074780-d0f7-11eb-9520-e5b2d52689e9.png)
![emoji_u1f4ce](https://user-images.githubusercontent.com/10349490/122639031-ef95bf00-d0f7-11eb-9d4e-e546fc514e58.png)

**Comment by C1710** [comment]
> 
> 
> > So, the Emoji13 branch should now contain all the assets for Emoji 13 (except for gendered mixed-skintone couples with the person with the lighter skincolor standing left).
> > But, I currently have an issue with the rendering, so it'll probably take at least until tomorrow, plus the EmojiCompat version which will be released a bit later.
> 
> Just FYI, this is how the emojis are currently rendered and I have absolutely no idea, why:
> ![emoji_u1f4a6](https://user-images.githubusercontent.com/10349490/122639026-e9074780-d0f7-11eb-9520-e5b2d52689e9.png)
> ![emoji_u1f4ce](https://user-images.githubusercontent.com/10349490/122639031-ef95bf00-d0f7-11eb-9d4e-e546fc514e58.png)
> 
> [EDIT] Now it's getting cursed:
> ![emoji_u264000ef](https://user-images.githubusercontent.com/10349490/122643333-86ba4100-d10f-11eb-9142-6042bb33de74.png)

Update: I still don't know what caused this bug, but using the minimal example provided for [resvg](https://github.com/RazrFalcon/resvg/) it works.  
However, I still have a problem with name parsing which is required for the Emoji 13 emojis I added.

**Comment by C1710** [comment]
A prerelease is now available in the [emoji13-branch](https://github.com/C1710/blobmoji/tree/emoji13):
https://github.com/C1710/blobmoji/blob/emoji13/fonts/Blobmoji.ttf

However:
- I haven't tested it yet
- The colors might be a little off for some emojis, due to the reduced color palette
- There is no EmojiCompat font available yet
- It is significantly larger (about 2MB) than the last version; this is because I didn't get [`imagequant`](https://github.com/ImageOptim/libimagequant) to work correctly yet
- Some flags are missing as they are aliases, but with the alias path being in the SVG files themselves, which I don't support as of now
- The screen text for the [technologist emojis](https://github.com/C1710/blobmoji/blob/emoji13/svg13/technologist.svg) isn't rendered for some reason
- There are some missing variations
- Currently, only Emoji 13.0 is supported

**Comment by C1710** [comment]
So far, everything seems to be fine except for the men in tuxedo.  
I forgot to generate the skintones for them.

But: Good news is that the EmojiCompat-font is also almost ready!

**Comment by C1710** [comment]
> 
> 
> So far, everything seems to be fine except for the men in tuxedo.
> I forgot to generate the skintones for them.
> 
> But: Good news is that the EmojiCompat-font is also almost ready!

It's now included

**Comment by choukawaii** [comment]
> > So far, everything seems to be fine except for the men in tuxedo.
> > I forgot to generate the skintones for them.
> > But: Good news is that the EmojiCompat-font is also almost ready!
> 
> It's now included

Waiting for the release. 😁

**Comment by C1710** [comment]
> 
> 
> > > So far, everything seems to be fine except for the men in tuxedo.
> > > I forgot to generate the skintones for them.
> > > But: Good news is that the EmojiCompat-font is also almost ready!
> > 
> > 
> > It's now included
> 
> Waiting for the release. 😁

You can already use it, so you actually don't really need to wait for a release :D
https://github.com/c1710/blobmoji/tree/emoji13

Normal font: https://github.com/C1710/blobmoji/blob/emoji13/fonts/Blobmoji.ttf
EmojiCompat font: https://github.com/C1710/blobmoji/blob/emoji13/emojicompat/BlobmojiCompat.ttf

**Comment by C1710** [comment]
I just figured out that there are about 200 missing emojis (although some aliases and other false positives are included in that list and it's mostly skin tones or ungendered versions, so nothing what actually _breaks_ everything).

You can find the list here: https://gist.github.com/C1710/0a9fc330f5e86a8b007c814d81c68992

**Comment by C1710** [comment]
> 
> 
> I just figured out that there are about 200 missing emojis (although some aliases and other false positives are included in that list and it's mostly skin tones or ungendered versions, so nothing what actually _breaks_ everything).
> 
> You can find the list here: https://gist.github.com/C1710/0a9fc330f5e86a8b007c814d81c68992

Most of them are now included.  
Still missing/wrong(ly gendered)/I'm not sure:
https://gist.github.com/C1710/0a9fc330f5e86a8b007c814d81c68992

**Comment by C1710** [comment]
For those of you who can live without skin toned kisses and mixed skin tone kisses and couples with heart, you can now also try the Emoji 13.1 version: https://github.com/C1710/blobmoji/tree/emoji13-1

**Comment by C1710** [comment]
ToDo:
- [ ] Add missing gender-neutral emojis
- [ ] Add missing skin tones
- [ ] Add missing flags/flag aliases (possibly even with their own flags)

**Comment by thayerfox** [comment]
Thank you for taking the time to update this!  I am thrilled the blobs live on.

(Since simonsmh archived his magisk module, and since the replacement is also dead, I updated his old module with your new font, for those who wish to still use magisk to install this: https://github.com/thayerfox/magisk-notocoloremoji-replacer/releases/download/13.1_pre/Blobmoji_13_1pre.zip)

**Comment by DavidBerdik** [comment]
> and since the replacement is also dead

Not to worry! It is not dead. I just want to wait for a final release before I rebuild it.

**Comment by thayerfox** [comment]
> > and since the replacement is also dead
> 
> Not to worry! It is not dead. I just want to wait for a final release before I rebuild it.

I was referring to https://github.com/Magisk-Modules-Repo/Fontchanger (I didn't see your fork either)

**Comment by DavidBerdik** [comment]
> I was referring to https://github.com/Magisk-Modules-Repo/Fontchanger (I didn't see your fork either)

Ah okay. My bad. My latest version (https://github.com/DavidBerdik/blobmoji-font/releases/tag/blobmoji-13.1-pre) is a mashup of non-blob versions for scenarios in which a new version was not already available here so yours is still more complete. 😁

**Comment by C1710** [comment]
So, it should be complete now, with all (mixed-)skintoned emojis and genderless versions for all(?) Emoji 5-emojis...

However, everything was a bit chaotic (partially thanks to Affinity Designer corrupting some SVGs, partially thanks to me having a typo in `couple with heart` (I wrote "hearts") and many other skintone and gender-stuff), so it might be possible that there are some glitches, missing emojis, etc.

**Comment by C1710** [comment]
So there are currently 32 emojis from before Emoji 5 that are not un-gendered.

These are the ones which I consider most important as they show a person's face (which is, except usually for the color of their clothing, the only thing that distinguishes the genders) and are rather general in their use:
- [ ] Person Tipping Hand
- [ ] Face Massage
- [ ] Face with No Good Gesture 
- [ ] Face with Ok Gesture
- [ ] Happy Person Raising One Hand
- [ ] Person Frowning
- [ ] Person with Pouting Face
- [ ] Face Palm
- [ ] Shrug


These are the ones of which I would personally say that it would be okay to release 13.1 without them:
- [ ] Runner
- [ ] Surfer
- [ ] Swimmer
- [ ] Weight Lifter
- [ ] Family
- [ ] Police Officer
- [ ] Person with Bunny ears
- [ ] Person with Turban
- [ ] Construction Worker
- [ ] Guardsperson
- [ ] Haircut
- [ ] Sleuth or Spy
- [ ] Person Bowing Deeply
- [ ] Rowboat
- [ ] Bicyclist
- [ ] Mountain Bicyclist
- [ ] Pedestrian
- [ ] Person doing Cartwheel
- [ ] Juggling
- [ ] Wrestlers
- [ ] Water Polo
- [ ] Handball
- [ ] Person with Ball

**Comment by C1710** [comment]
With the missing genderless emojis added, I think we're good to go...

Validation doesn't report missing emojis (apart from some confusion with VS-16 being or not being present in some cases) and I think there are no glitches/names with unrecognized names anymore.

**Comment by C1710** [comment]
I will probably release it this saturday. If you can't remember the date, just look on one of the calendar emojis!

**Comment by C1710** [comment]
Since I've now verified that all emojis are there and also now work with EmojiCompat (with some tricks, as the font doesn't include number characters anymore), I'll close this issue.

You can (as of now) download the font on the [`emoji13-1`-branch](https://github.com/C1710/blobmoji/tree/emoji13-1/fonts), which will (unless there are really absolutely-last-minute changes) be the final font to be released (and merged into master) on Saturday.

**Comment by DavidBerdik** [comment]
Operating under the assumption that the build on the `emoji13-1` branch is probably not going to receive any last-minute changes before release on Saturday, I have updated my fork of the Blobmoji Magisk module. Anyone interested can download it here: https://github.com/DavidBerdik/blobmoji/releases/tag/13.1

**Comment by C1710** [comment]
> 
> 
> Operating under the assumption that the build on the `emoji13-1` branch is probably not going to receive any last-minute changes before release on Saturday, I have updated my fork of the Blobmoji Magisk module. Anyone interested can download it here: https://github.com/DavidBerdik/blobmoji/releases/tag/13.1

Did you use the one in the `fonts` directory or the one from the release page? (I haven't updated that one for a while... 😬 But if you used that one, I would suggest you wait until tomorrow with an update)

**Comment by DavidBerdik** [comment]
I used the `Blobmoji.ttf` file attached to https://github.com/C1710/blobmoji/releases/tag/v2021-07-12-Emoji13.1-pre

Is that correct?

**Comment by C1710** [comment]
> 
> 
> I used the `Blobmoji.ttf` file attached to https://github.com/C1710/blobmoji/releases/tag/v2021-07-12-Emoji13.1-pre
> 
> Is that correct?

There have been some changes/fixes since then (it still works, almost all the normal emojis are fine):  
- The colored squares still have a different size to the b/w ones (not that problematic)
- Some kiss emojis have the right one in red (that's the placeholder color I use for multi-skintone generation)
- The black heart isn't updated yet to match the other hearts (also not that dramatic)
- The man in the manual wheelchair is transgender or something (the woman-version is shown)

But: Please just wait until the release tomorrow

**Comment by DavidBerdik** [comment]
Under normal circumstances, I would have waited until tomorrow, but I'm leaving for vacation tomorrow and wanted to build the new emoji set before then (I'm impatient, I guess). Assuming I don't get bored and rebuild the module when I'm on vacation, I'll make sure to do it when I return!

**Comment by DavidBerdik** [comment]
Also, I have updated the download page for the module to indicate that a second release will be necessary at a later time.

**Comment by C1710** [comment]
> Under normal circumstances, I would have waited until tomorrow, but I'm leaving for vacation tomorrow and wanted to build the new emoji set before then (I'm impatient, I guess). Assuming I don't get bored and rebuild the module when I'm on vacation, I'll make sure to do it when I return!

Well, in that case I wish you a great vacation! 😄

**Comment by DavidBerdik** [comment]
Thank you! 😁

**Comment by C1710** [comment]
> 
> 
> Also, I have updated the download page for the module to indicate that a second release will be necessary at a later time.

Release is now done for Emoji 13.1 and 14.0 😊

**Comment by DavidBerdik** [comment]
I built a new version of the module using the 14.0 release. I didn't bother with 13.1.

https://github.com/DavidBerdik/blobmoji/releases/tag/14-r1
